docs/checklist_applicationclass_2-markdown_v1.1.md

```diff
@@ -0,0 +1,157 @@
+
+**EÄM.6** - **[ok]** A detailed change history (change log) exists providing information about the functionalities and bug fixes of a software version. *(from application class 2)*
+
+**EÄM.7** - **[ok]** A repository is set up in a version control system. The repository is adequately structured and ideally contains all artefacts for building a usable software version and for testing it. *(from application class 1)*
+
+**EÄM.8** - **[ok]** Every change of the repository ideally serves a specific purpose, contains an understandable description and leaves the software in a consistent, working state. *(from application class 1)*
+
+**EÄM.9** - **[ok]** If there are multiple common development branches, their purpose can be identified easily. *(from application class 2)*
+
+## Design and Implementation <a name="design-implementierung"></a>
+**EDI.1** - **[ok]** The usual patterns and solution approaches of the selected programming language are used and a set of rules regarding the programming style is consistently applied. The set of rules refers at least to the formatting and commenting. *(from application class 1)*
+
+**EDI.2** - **[ok]** The software is structured modularly as far as possible. The modules are coupled loosely. I.e., a single module depends as little as possible on other modules. *(from application class 1)*
+
+**EDI.3** - **[todo]** Ideally, there are module tests for every module. The module tests demonstrate their typical use and constraints. *(from application class 2)*
+
+**EDI.4** - **[ok]** The implementation reflects the software architecture. *(from application class 2)*
+
+**EDI.5** - **[ok]** It is continuously paid attention to room for improvement during development. Required changes (refactoring) may be implemented directly or prioritised through the change process. *(from application class 2)*
+
+**EDI.6** - **[ok]** Suitability of rules with respect to the programming style is checked regularly. The preferred approaches and design patterns, relevant design principles and rules, as well as rules for permitted and non-permitted language elements may be supplemented. *(from application class 2)*
+
+**EDI.7** - **[ok]** Adherence to simple rules concerning the programming style is checked or ensured automatically. *(from application class 2)*
+
+**EDI.8** - **[todo]** Key design principles are defined and communicated. *(from application class 2)*
+
+**EDI.9** - **[ok]** The source code and the comments contain as little duplicated information as possible. ("Don`t repeat yourself.") *(from application class 1)*
+
+**EDI.10** - **[ok]** Prefer simple, understandable solutions. ("Keep it simple and stupid."). *(from application class 1)*
+
+## Software Test <a name="software-test"></a>
+**EST.1** - **[todo]** An overall test strategy is coordinated and defined. It is checked regularly for appropriateness. *(from application class 2)*
```

Comment:
I think we have an overall test strategy, but we simply need more testing

Reply:
```suggestion
**EST.1** - **[ok]** An overall test strategy is coordinated and defined. It is checked regularly for appropriateness. *(from application class 2)*
```